apps/build_memory_index.cpp

```diff
@@ -120,32 +120,34 @@ int main(int argc, char **argv)
         size_t data_num, data_dim;
         diskann::get_bin_metadata(data_path, data_num, data_dim);
 
+        auto index_build_params = diskann::IndexWriteParametersBuilder(L, R)
+                                      .with_filter_list_size(Lf)
+                                      .with_alpha(alpha)
+                                      .with_saturate_graph(false)
+                                      .with_num_threads(num_threads)
+                                      .build();
+
+        auto build_params = diskann::IndexBuildParamsBuilder(index_build_params)
```

Comment:
whats the difference between index_build_params and build_params? if it is the filtering addons, then shouldnt Lf go into build_params?

Reply:
index_build_params aka (Indexwriteparams) are tunable params for building index like (L, R... etc) the BuildParams in turn  contains the IndexWritePrams and files path's like label_file, data_file, tag_file etc. BuildParams is to hold all those fields, as we have consolidated build method.